multiuser/api/che-multiuser-api-authentication-commons/src/test/java/org/eclipse/che/multiuser/api/authentication/commons/token/HeaderRequestTokenExtractorTest.java

```diff
@@ -0,0 +1,65 @@
+/*
+ * Copyright (c) 2012-2018 Red Hat, Inc.
+ * This program and the accompanying materials are made
+ * available under the terms of the Eclipse Public License 2.0
+ * which is available at https://www.eclipse.org/legal/epl-2.0/
+ *
+ * SPDX-License-Identifier: EPL-2.0
+ *
+ * Contributors:
+ *   Red Hat, Inc. - initial API and implementation
+ */
+package org.eclipse.che.multiuser.api.authentication.commons.token;
+
+import static javax.ws.rs.core.HttpHeaders.AUTHORIZATION;
+import static org.mockito.ArgumentMatchers.eq;
+import static org.mockito.Mockito.when;
+import static org.testng.Assert.assertEquals;
+
+import javax.servlet.http.HttpServletRequest;
+import javax.ws.rs.BadRequestException;
+import org.mockito.Mock;
+import org.mockito.testng.MockitoTestNGListener;
+import org.testng.annotations.DataProvider;
+import org.testng.annotations.Listeners;
+import org.testng.annotations.Test;
+
+@Listeners(MockitoTestNGListener.class)
+public class HeaderRequestTokenExtractorTest {
+
+  private HeaderRequestTokenExtractor tokenExtractor = new HeaderRequestTokenExtractor();
+
+  @Mock HttpServletRequest servletRequest;
+
+  @Test(dataProvider = "validHeadersProvider")
+  public void shouldExtractTokensFromValidHeaders(String headerValue, String expectedToken) {
+
+    when(servletRequest.getHeader(eq(AUTHORIZATION))).thenReturn(headerValue);
+
+    // when
+    String token = tokenExtractor.getToken(servletRequest);
+
+    // then
+    assertEquals(token, expectedToken);
+  }
+
+  @Test(
+      expectedExceptions = BadRequestException.class,
+      expectedExceptionsMessageRegExp = "Invalid authorization header format.")
+  public void shouldThrowExceptionOnInvalidToken() {
+
+    when(servletRequest.getHeader(eq(AUTHORIZATION))).thenReturn("bearertoken123");
+
+    // when
+    tokenExtractor.getToken(servletRequest);
+  }
+
+  @DataProvider
+  private Object[][] validHeadersProvider() {
+    return new Object[][] {
+      {"token123", "token123"},
+      {"bearer token123", "token123"},
+      {"bearer       token123", "token123"},
```

Comment:
are you sure this really valid?

Reply:
Integresting question.  OAuth 2.0 Specification says `"Bearer" 1*SP b64token`, but its probably not a big problem if it's 2 spaces...? Or it is ?

Reply:
I don't know. I feel like we should follow spec, unless we know exactly why it can be other way. And currently we don't know what a why is it happening. What I'm pretty sure is that the `ArrayIndexOutOfBoundsException` was not caused by extra space, because `System.out.println("a  b".split(" ").length);` gives length `3`...

Reply:
yes, must probaby it just has no any spaces

Reply:
ok will made it only accepting 1 space

Reply:
BTW: `System.out.println("a  b".split(" ").length); // 3`  try it and you will be surprized :)

Reply:
github doesn't handle spaces really well, so I don't know what you mean exactly :)

Reply:
https://gist.github.com/sparkoo/06832198edbe1ac9e59acc36a8bce9a7